homeassistant/components/weather/metoffice.py

```diff
@@ -0,0 +1,117 @@
+"""
+Support for UK Met Office weather service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.metoffice/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.weather import WeatherEntity, PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, TEMP_CELSIUS
+from homeassistant.helpers import config_validation as cv
+# Reuse data and API logic from the sensor implementation
```

Comment:
We created the `weather` component to replace the `sensor` platforms for meteorological data.

So, I tend to say that the `weather` component should be the main implementation because there will be a frontend element some day and the `sensor` platforms will be obsolete. But as we are moving slowing in this regard I will leave it up to you to decide what platform should be the main implementation.